### PR TITLE
Move "ensure-up-to-date" execution to CircleCI

### PR DESCRIPTION
**Pull request by jmini**
### PR checklist

- [ ] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [ ] Ran the shell script under `./bin/` to update Petstore sample so that CIs can verify the change. (For instance, only need to run `./bin/{LANG}-petstore.sh` and `./bin/security/{LANG}-petstore.sh` if updating the {LANG} (e.g. php, ruby, python, etc) code generator or {LANG} client's mustache templates). Windows batch files can be found in `.\bin\windows\`.
- [ ] Filed the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`, `3.3.x`, `4.0.x`. Default: `master`.
- [ ] Copied the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) to review the pull request if your PR is targeting a particular programming language.

### Description of the PR

Extracted from PR #929:
@wing328 proposed to move execution of `ensure-up-to-date` (see #80) to CircleCI.

This PR does the change and I will perform some checks to ensure that it works as expected.

### CODE REVIEW

File: CI/circle_parallel.sh [hunk 1]
```diff
@@ -8,6 +8,10 @@ if [ "$NODE_INDEX" = "1" ]; then
   cp CI/pom.xml.circleci pom.xml
   java -version
   mvn --quiet verify -Psamples
+elif [ "$NODE_INDEX" = "2" ]; then
+  echo "Running node $NODE_INDEX to test ensure-up-to-date"
+  mvn clean package -DskipTests -Dmaven.javadoc.skip=true
```
Comment (wing328) [inline CI/circle_parallel.sh]: "mvn clean install" was run before running this script (https://github.com/OpenAPITools/openapi-generator/pull/944/files#diff-29944324a3cbf9f4bd0162dfe3975d88R69) so I don't think we need to do it again or I probably miss the reason behind.

Comment (jmini) [reply CI/circle_parallel.sh]: It is just a matter of having `modules/openapi-generator-cli/target/openapi-generator-cli.jar`. I have put here the command that is sufficient to produce it (with a clean to be sure that it gets updated).

Comment (jmini) [reply CI/circle_parallel.sh]: On shippable, there was already a `mvn clean install`, so what I did is just moving it before `ensure-up-to-date` with the initial commit:

https://github.com/OpenAPITools/openapi-generator/commit/00f176ad956c06fc6027b4b0e22d982b928552c6#diff-033d042b39454e8e066ab38fd89610a5

**Comment by jmini** [comment]
Commit df18063 added to see if ensure-up-to-date is working. CircleCI should be red.

**Comment by jmini** [comment]
Error as expected:

circleci

Important: you need to switch to the second tab `#queue-placeholder/containers/2` in the URL to see the error.

File: CI/circle_parallel.sh (hunk 1 above)
Comment (wing328) [reply CI/circle_parallel.sh]: 👍

**Pull request merged**